projects/go-lib/src/lib/components/go-timepicker/go-timepicker.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { generateId } from '../../utilities/form.utils';
+
+@Component({
+  selector: 'go-timepicker',
+  styleUrls: ['./go-timepicker.component.scss'],
+  templateUrl: './go-timepicker.component.html',
+})
+
+export class GoTimepickerComponent implements OnInit {
+  displayAbove: boolean = false;
+  displayFromRight: boolean = true;
+  timeOpen: boolean = false;
+  id: string;
+  openTimeValue: string;
+
+  @Input() key: string;
+  @Input() label: string;
+  @Input() control: FormControl;
+  @Input() hints: string[];
+  @Input() placeholder: string = '';
+  @Input() theme: string = 'light';
+  @Input() appendToContent: boolean = false;
+
+  selectedTime: string;
+
+  @ViewChild('datepickerInput', { static: true }) datepickerInput: ElementRef;
+
+  ngOnInit(): void {
+    this.id = this.key || generateId(this.label, 'timepicker');
+    this.selectedTime = this.changeTimeFormat(this.control.value);
+  }
+
+  public changeTimeFormat(timeString: string): string {
+    if (!timeString) {
+      return;
+    }
+    const H: number = +timeString.substr(0, 2);
+    const h: number | string = H % 12 || 12;
+    const hour: number | string = h > 9 ? h : '0' + h;
+    const ampm: string = H < 12 || H === 24 ? 'AM' : 'PM';
+    timeString = hour + timeString.substr(2, 3) + ' ' + ampm;
+    return timeString;
+  }
+
+  public toggleTimepicker(event: Event): void {
+    event.stopPropagation();
+    // Have to disable this here because of the event that we need to stop propagation on.
+    if (this.control.disabled) {
+      return;
+    }
+
+    this.timeOpen = !this.timeOpen;
+    const position: string = 'top';
+    const distance: object = this.datepickerInput.nativeElement.getBoundingClientRect();
+    this.displayAbove =
+      window.innerHeight - distance[position] < 350 && !this.appendToContent;
+    const convert24Hr: string = /^\s*$/.test(this.selectedTime)
+      ? undefined
+      : this.convertTo24Hour(this.selectedTime);
+    this.openTimeValue = convert24Hr ? convert24Hr : undefined;
+  }
+
+  public timePicked(event: any): void {
+    if (event) {
+      const selctedTime: string =
+        event.hours + ':' + event.minutes + ' ' + event.ampm;
+      this.control.setValue(this.convertTo24Hour(selctedTime));
+      this.selectedTime = selctedTime;
+    } else {
+      this.selectedTime = '';
```

Comment:
Should we also null out the control value here as well?

Reply:
I have added this code `this.selectedTime = null` and also ` selectedTime: string = null` in declaration part. Is it right which you want?

Reply:
Not exactly. I'm suggesting setting `this.control.setValue(null)` or we could even just move the following line `this.control.setValue(this.convertTo24Hour(selctedTime));` after the if else statement. The reason for this is because even though setting `this.selectedTime = ''` here would empty out the select, it wouldn't change the value that is actually in the form, so when the form is submitted, instead of sending an empty value it would send the last selected value, which doesn't seem correct.

Reply:
Yes. You are right.  I have updated that code.